examples/howto/data_container.myst.md

```diff
@@ -11,43 +11,103 @@ kernelspec:
 ---
 
 (data_container)=
-# Using shared variables (`Data` container adaptation)
+# Using Data Containers
 
 :::{post} Dec 16, 2021
 :tags: posterior predictive, shared data 
 :category: beginner
-:author: Juan Martin Loyola, Kavya Jaiswal, Oriol Abril
+:author: Juan Martin Loyola, Kavya Jaiswal, Oriol Abril, Jesse Grabowski
 :::
 
 ```{code-cell} ipython3
 import arviz as az
 import matplotlib.pyplot as plt
 import numpy as np
 import pandas as pd
-import pymc3 as pm
+import pymc as pm
 
 from numpy.random import default_rng
 
-print(f"Running on PyMC3 v{pm.__version__}")
+print(f"Running on PyMC v{pm.__version__}")
 ```
 
 ```{code-cell} ipython3
 %matplotlib inline
 %config InlineBackend.figure_format = 'retina'
-RANDOM_SEED = 8927
+RANDOM_SEED = sum(map(ord, "Data Containers in PyMC"))
 rng = default_rng(RANDOM_SEED)
 az.style.use("arviz-darkgrid")
 ```
 
-## The Data class
+## Introduction
 
-The {class}`pymc.Data` container class wraps the theano shared variable class and lets the model be aware of its inputs and outputs. This allows one to change the value of an observed variable to predict or refit on new data. All variables of this class must be declared inside a model context and specify a name for them.
+After building the statistical model of your dreams, you're going to need to feed it some data. Data is "shown" to a PyMC model via the `observed` keyword argument of all random variables. This argument accepts a wide variety of datatypes, including numpy `ndarrays`, pandas `Series` and `DataFrame`, and even pytensor `TensorVariables`. Although you can pass these "raw" datatypes to your PyMC model, the best way is to use one of two {class}`pymc.Data` containers. 
 
-In the following example, this is demonstrated with fictional temperature observations.
+These containers make it extremely easy to work with data in a PyMC model. They offer a range of benefits, including:
+
+1. Visualization of data as a component of your probabilistic graph
+2. Access to labeled dimensions for readability and accessibility
+3. Support for swapping out data for out-of-sample prediction, interpolation/extrapolation, forcasting, etc.
+
+This notebook will illustrate each of these benefits in turn, and show you the best way to integrate data into your PyMC modeling workflow. 
+
++++
+
+## Types of Data Containers
+
+ PyMC offers two data containers, depending on your needs: {class}`pymc.ConstantData` and {class}`MutableData`. Both will help you visualize how data fits into your model, store the data in an `InfereceData` for reproducibility, and give access to labeled dimenions. As the names suggest, however, only `MutableData` allows you to change your data to do out-of-sample inference tasks. This ability does, however, come with a small performance cost.
```

Comment:
It also allows you to use the same model for different datasets, which can be useful e.g., when doing parameter recovery studies or sensitivity analysis